### ISSUE TITLE
What are A and N in B N A T?

### ISSUE THREAD

**Issue by gyuseok0917** [935963004/LaBraM#34]
`
class TemporalConv(nn.Module):
    """ Image to Patch Embedding
    """
    def __init__(self, in_chans=1, out_chans=8):
        super().__init__()
        self.conv1 = nn.Conv2d(in_chans, out_chans, kernel_size=(1, 15), stride=(1, 8), padding=(0, 7))
        self.gelu1 = nn.GELU()
        self.norm1 = nn.GroupNorm(4, out_chans)
        self.conv2 = nn.Conv2d(out_chans, out_chans, kernel_size=(1, 3), padding=(0, 1))
        self.gelu2 = nn.GELU()
        self.norm2 = nn.GroupNorm(4, out_chans)
        self.conv3 = nn.Conv2d(out_chans, out_chans, kernel_size=(1, 3), padding=(0, 1))
        self.norm3 = nn.GroupNorm(4, out_chans)
        self.gelu3 = nn.GELU()

    def forward(self, x, **kwargs):
        x = rearrange(x, 'B N A T -> B (N A) T')
        B, NA, T = x.shape
        x = x.unsqueeze(1)
        x = self.gelu1(self.norm1(self.conv1(x)))
        x = self.gelu2(self.norm2(self.conv2(x)))
        x = self.gelu3(self.norm3(self.conv3(x)))
        x = rearrange(x, 'B C NA T -> B NA (T C)')
        return x
`

Hello. I am analyzing your code to utilize it, and in the forward method of the **TemporalConv** class in **modeling_pretrain.py** above, in the part where einops is rearranged, the input dimension is listed as 4-dimensional. I thought B is the batch size, N is the number of electrodes, and T is the sample length, but I couldn't figure out what A means. Also, I have a question about whether N is the number of electrodes because of A.

**Comment by 935963004** [comment]
A is the time length. For example, a 4-second EEG sample with 64 channels has the shape of (B, 64, 4, 200).

**Comment by gyuseok0917** [comment]
Hello. First of all, thank you for answering my question. After that, I am writing because I have some more questions while analyzing the code.

1. When merging B, N, A, T to B, (N A), T using rearrange, why are you merging the EEG electrode count (N) and time seconds (A) dimensions?
2. Are the parameters of "TemporalConv" class applied according to the number of electrodes and time length of the data you used?
